pandas/core/groupby/generic.py

```diff
@@ -1569,6 +1574,198 @@ def func(df):
+                keys: Any = []
+                remaining_columns = self._selected_obj.columns
+            elif isinstance(self._selected_obj, Series):
+                keys = [grouping.name for grouping in self.grouper.groupings]
+                remaining_columns = [self._selected_obj.name]
+            else:
+                if isinstance(self.keys, ops.BaseGrouper):
+                    keys = [grouping.name for grouping in self.keys.groupings]
+                elif isinstance(self.keys, str):
+                    keys = [self.keys]
+                else:
+                    keys = self.keys
+
+                remaining_columns = [
+                    key for key in self._selected_obj.columns if key not in keys
+                ]
+
+            if subset is not None:
+                for key in subset:
+                    if key in keys:
+                        raise ValueError(
+                            f"Key {key} in subset cannot be one of "
+                            "the groupby column keys"
+                        )
+                remaining_columns = subset
+
+            if dropna:
+                df = df.dropna(subset=remaining_columns, axis="index", how="any")
+
+            # Add the remaining_column keys to the main grouper
+            main_grouper, _, _ = get_grouper(
+                df,
+                key=self.keys,
+                axis=self.axis,
+                level=self.level,
+                sort=self.sort,
+                mutated=self.mutated,
+                dropna=self.dropna,
+            )
+            groupings = list(main_grouper.groupings)
+            for key in remaining_columns:
+                grouper, _, _ = get_grouper(
+                    df,
+                    key=key,
+                    axis=self.axis,
+                    level=self.level,
+                    sort=self.sort,
+                    mutated=self.mutated,
+                    # nulls have already been dropped from remaining_columns
+                    dropna=False,
+                )
+                groupings += list(grouper.groupings)
+
+            # Take the size of the overall columns
+            result = df.groupby(
+                groupings,
+                sort=self.sort,
+                observed=self.observed,
+                dropna=self.dropna,
+            ).size()
+            result.name = RESULT_NAME
+
+            if normalize:
+                # Normalize the results by dividing by the original group sizes
+                indexed_group_size = df.groupby(
+                    main_grouper,
+                    sort=self.sort,
+                    observed=self.observed,
+                    dropna=self.dropna,
+                ).size()
+                if non_column_grouping:
+                    # The common index needs a common name
+                    indexed_group_size.index.set_names("Group", inplace=True)
+                    result.index.set_names("Group", level=0, inplace=True)
```

Comment:
This fails if multiple not in_axis groupers are passed, e.g. `df.groupby([[0, 1, 2], [0, 1, 1]])`. Also not thrilled with using "Group" and then relying on alignment. Could that lead to failure if the user has a grouper / column named "Group"?

Reply:
> not thrilled with using "Group"
Changed to "_group_"